### ISSUE TITLE
Cannot register a new client key - Failed to set ACL entries: No mapping between account names and security IDs was done

### ISSUE THREAD

**Issue by steakpowered** [dechamps/LGTVDeviceListener#1]
I've tried executing for the first time but it keeps failing with this message:

FATAL: Failed to set ACL entries: No mapping between account names and security IDs was done.

1.  I see the confirmation dialog on my TV, so I know it's connecting.
2. I'm using Windows 11 with a microsoft account connected.
3. Tried with normal and admin powershell.
4. I searched the web and tried the fixes, but they all say it's caused when a username is changed.  However, I never changed my username.

Thanks in advance!

**Comment by dechamps** [comment]
Mmm, that's not supposed to happen. Basically that error message means that, while setting Windows ACLs on the client key file that LGTVDeviceListener is trying to write, [`SetEntriesInAcl()`](https://learn.microsoft.com/en-us/windows/win32/api/aclapi/nf-aclapi-setentriesinaclw) is returning `ERROR_NONE_MAPPED`.

Looking at the code I'm not quite sure why it would be working for me but not for you. Maybe it has to do with localization on the `"CREATOR OWNER"` name - is your Windows configured with a language other than English by any chance?

Another possibility is that `"NT SERVICE\LGTVDeviceListener"` is not a valid account name if a service was never created. I doubt it though.

I guess I should try to reproduce your problem in a fresh VM to double-check that it works on a clean system.

In the meantime, you should be able to work around the issue by creating the client key file yourself, as LGTVDeviceListener will not try to set ACLs on a file that already exists. Here's how:

1. Try to register as usual, but add the `--verbose` command line parameter.
2. Before LGTVDeviceListener fails, you should see something like the following in the output: `[verbose] Received message from LGTV: {"type":"registered","id":1,"payload":{"client-key":"XXX"}}`. `XXX` is your client key.
3. Create a new text file named `C:\ProgramData\LGTVDeviceListener.client-key` and copy-paste the key (just the key - not the surrounding JSON) into the file. Do not write anything else in the file (not even a newline).
4. You should then be able to proceed with next steps.

This workaround is a bit less secure than what LGTVDeviceListener does, because the resulting file doesn't have locked down ACLs to prevent random Windows users (e.g. background services) from reading it, but honestly this is quite benign in the grand scheme of things.

**Comment by dechamps** [comment]
Okay I'm able to reproduce in a fresh Windows 11 VM (where the Windows language happens to be French, but I have not verified if this matters). So this is definitely an LGTVDeviceListener bug, sorry about that. I'll figure out what's going on and fix it when I have some time.

**Comment by steakpowered** [comment]
1. Windows is configured as English.  
2. It does show that I am the owner of the file and my account is also an admin account.
3. Didn't get far enough to create a service last time.  I used the instructions above and was able to get a key. 
4. I had to make tweak to your instructions.  The parameter after --device-name had to be in quotes for it to work.
5. Anything I can do to help you debug, let me know.
6. When the device is added, it switches in about 1 second after the device is detected.  When the device is removed, it switches after 3 seconds. 

Overall, great work and this helps with using this as a KVM and being able to still maintain 4k 120hz on both inputs.  Thanks!

**Comment by dechamps** [comment]
The root cause is that LGTVDeviceListener is attempting to set an ACL on the client key file where the trustee of one of the ACEs is set to `NT SERVICE\LGTVDeviceListener`. What I didn't realize is that this only works if a service by that name (`LGTVDeviceListener`) already exists, otherwise [`SetEntriesInAcl()`](https://learn.microsoft.com/en-us/windows/win32/api/aclapi/nf-aclapi-setentriesinaclw) fails. I missed this during development, most likely because the service might have already been installed on my machine before I wrote the ACL code.

Nevertheless I did not expect this call to fail, given that the SID for a `NT SERVICE` account is a pure function of the service name, is not machine-specific and stays the same regardless of system state, so there's no reason why Windows would insist that the service already exist before resolving the SID. Clearly this is a limitation of `SetEntriesInAcl`; I'll see if there's a way to work around it.
